test/e2e/handler/nns_update_timestamp_test.go

```diff
@@ -17,33 +18,43 @@ import (
 var _ = Describe("[nns] NNS LastSuccessfulUpdateTime", func() {
 	var (
 		originalNNSs map[string]nmstatev1beta1.NodeNetworkState
-	)
-	BeforeEach(func() {
-		originalNNSs = map[string]nmstatev1beta1.NodeNetworkState{}
-		for _, node := range allNodes {
-			key := types.NamespacedName{Name: node}
-			originalNNSs[node] = nodeNetworkState(key)
+		nnsByNode    = func() map[string]nmstatev1beta1.NodeNetworkState {
+			NNSs := map[string]nmstatev1beta1.NodeNetworkState{}
+			for _, node := range allNodes {
+				key := types.NamespacedName{Name: node}
+				NNSs[node] = nodeNetworkState(key)
+			}
+			return NNSs
 		}
-	})
+	)
 	Context("when network configuration hasn't change", func() {
 		It("should not be updated", func() {
-			for node, originalNNS := range originalNNSs {
-				// Give enough time for the NNS to be updated (3 interval times)
-				timeout := 3 * nmstatenode.NetworkStateRefresh
-				key := types.NamespacedName{Name: node}
+			Eventually(func() error {
+				originalNNSs = nnsByNode()
+				By("Give enough time for the NNS Reconcile to happen (3 interval times)")
+				time.Sleep(3 * nmstatenode.NetworkStateRefresh)
```

Comment:
Why do we need this on top of the eventually?

Why is 2 interval times not enough?

Reply:
This Eventually just repeting the test since sometimes there are change at network level not done by a NNCP and NNS get updated so we are trying to run this test three times until it works (yes I don't like it).

Reply:
Reduced to two interval time, don't know why I got 3.